### PR TITLE
[Depends] Dependency updates for 0.14.0

### PR DESCRIPTION
**Pull request by fanquake**
This includes:
Boost 1.61.0 -> [1.63.0](http://www.boost.org/users/history/version_1_63_0.html)
ccache 3.3.1 -> [3.3.3](https://ccache.samba.org/releasenotes.html#_ccache_3_3_3)
dbus 1.8.6 -> [1.10.14](https://cgit.freedesktop.org/dbus/dbus/tree/NEWS?h=dbus-1.10)
Freetype 2.6.3 -> [2.7.1](https://sourceforge.net/projects/freetype/files/freetype2/2.7.1/)
native_ds_store c80c23706eae -> 1.1.0

QT, libevent and ZeroMQ (#9254) will be updated in separate PRs, as they will likely require further discussion.

### CODE REVIEW

File: depends/config.guess [hunk 1]
```diff
@@ -1453,4 +1459,4 @@ exit 1
 # time-stamp-start: "timestamp='"
 # time-stamp-format: "%:y-%02m-%02d"
 # time-stamp-end: "'"
-# End:
+# End:
```
Comment (paveljanik) [inline depends/config.guess]: Did they really deleted final newline? I do not believe so...

File: depends/config.sub [hunk 2]
```diff
@@ -1812,4 +1825,4 @@ exit
 # time-stamp-start: "timestamp='"
 # time-stamp-format: "%:y-%02m-%02d"
 # time-stamp-end: "'"
-# End:
+# End:
```
Comment (paveljanik) [inline depends/config.sub]: Ditto

File: depends/patches/native_biplist/sorted_list.patch [hunk 3]
```diff
@@ -1,6 +1,6 @@
---- a/biplist/__init__.py	2014-10-26 19:03:11.000000000 +0000
-+++ b/biplist/__init__.py	2016-07-19 19:30:17.663521999 +0000
-@@ -541,7 +541,7 @@
+--- a/biplist/__init__.py
++++ b/biplist/__init__.py
```
Comment (paveljanik) [inline depends/patches/native_biplist/sorted_list.patch]: No change i the diff, so why update it? Just to apply cleanly?

**Comment by paveljanik** [comment]
Concept ACK

**Comment by paveljanik** [comment]
```
/usr/bin/python3.4 ../contrib/macdeploy/custom_dsstore.py "dist/.DS_Store" "Bitcoin-Core"
Traceback (most recent call last):
  File "../contrib/macdeploy/custom_dsstore.py", line 7, in <module>
    from ds_store import DSStore
  File "/home/travis/build/bitcoin/bitcoin/depends/x86_64-apple-darwin11/native/lib/python/dist-packages/ds_store/__init__.py", line 1, in <module>
    from .store import DSStore, DSStoreEntry
  File "/home/travis/build/bitcoin/bitcoin/depends/x86_64-apple-darwin11/native/lib/python/dist-packages/ds_store/store.py", line 9, in <module>
    import mac_alias
  File "/home/travis/build/bitcoin/bitcoin/depends/x86_64-apple-darwin11/native/lib/python/dist-packages/mac_alias/__init__.py", line 2, in <module>
    from .bookmark import *
  File "/home/travis/build/bitcoin/bitcoin/depends/x86_64-apple-darwin11/native/lib/python/dist-packages/mac_alias/bookmark.py", line 295
    print 'Unknown data type %08x' % typecode
                                 ^
SyntaxError: Missing parentheses in call to 'print'
make: *** [dist/.DS_Store] Error 1
```

File: depends/config.guess (hunk 1 above)
Comment (fanquake) [inline depends/config.guess]: @paveljanik If you look at either file you'll see neither has a newline.

File: depends/patches/native_biplist/sorted_list.patch (hunk 3 above)
Comment (theuni) [inline depends/patches/native_biplist/sorted_list.patch]: Looks like it's just offset. Is no problem.

**Comment by theuni** [comment]
Looks like mac_alias needs new fixups for python3. I'd say just skip that bump for now, since the author doesn't seem to be in any hurry to update.

**Comment by theuni** [comment]
utACK without the mac_alias bump.

File: depends/config.guess (hunk 1 above)
Comment (paveljanik) [inline depends/config.guess]: Upstream, both has newlines.

File: depends/patches/native_biplist/sorted_list.patch (hunk 3 above)
Comment (paveljanik) [inline depends/patches/native_biplist/sorted_list.patch]: There is no problem even without this hunk 8) It is there to apply cleanly. No other purpose. And as such, it is useless IMO.

**Comment by fanquake** [comment]
Dropped the mac_alias commit, and fixed the config.guess/sub newline issue.

**Comment by jtimon** [comment]
Concept ACK

File: depends/config.guess [hunk 4]
```diff
@@ -1032,6 +1035,9 @@ EOF
     ppcle:Linux:*:*)
 	echo powerpcle-unknown-linux-${LIBC}
 	exit ;;
+    riscv32:Linux:*:* | riscv64:Linux:*:*)
```
Comment (laanwj) [inline depends/config.guess]: riscv32 yay :)

**Comment by laanwj** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/9468/commits/7f1fa99a216d41bc3cf96ff3d5e6185897af1fa9. I think the WIP tag can go?

**Pull request merged**